src/main/java/fi/helsinki/cs/tmc/core/domain/Course.java

```diff
@@ -67,6 +71,39 @@ public void setExercises(List<Exercise> exercises) {
         this.exercises = exercises;
     }
 
+    public List<Theme> getThemes() {
+        return themes;
+    }
+
+    public void generateThemes() {
+        themes = new ArrayList<>();
+        Map<String, Theme> themeMap = new HashMap<>();
+        for (Exercise ex : exercises) {
+            addExerciseToTheme(themeMap, ex);
+        }
+    }
+
+    private void addExerciseToTheme(Map<String, Theme> themeMap, Exercise ex) {
+        String themeName = ex.getName().split("-")[0];
+        Theme theme = themeMap.get(themeName);
+        if (theme == null) {
+            theme = new Theme(themeName);
```

Comment:
@nygrenh Are we still targeting Java 7? If not, I'd rather use `getOrDefault` from Java 8 for things like this.